addons/account/models/partner.py

```diff
@@ -235,7 +239,7 @@ def _get_fiscal_position(self, partner, delivery=None):
             return manual_fiscal_position
 
         # First search only matching VAT positions
-        vat_required = bool(partner.vat)
+        vat_required = self._get_vat_required(delivery, company)
```

Comment:
Maybe instead of `vat_required`, we should name this `vat_valid`.
Rationale is:
* a fiscal position may require a valid VAT (`vat_require`)
* a partner may have a valid VAT (`vat_valid`)
* When we search for the fiscal position, we only consider those with `vat_require` if `vat_valid==True`

The previous wording always confused me. I think that this PR would be easier to read too, especially the inheritance of ~~`_get_vat_required`~~ `_get_vat_valid` in `base_vat.

Reply:
I agree strongly on this one, I had just figured that the reason the wording `vat_require` was confusing me was that I didn't comprehend it properly